code/modules/hydroponics/compost_bin.dm

```diff
@@ -0,0 +1,192 @@
+#define BASE_BIOMASS_CAPACITY 1500
+#define BASE_COMPOST_CAPACITY 1500
+#define SOIL_COST 25
+#define SOIL /obj/item/stack/sheet/soil
+#define DECAY 0.2
+#define MIN_CONVERSION 10
+/**
+  * # compost bin
+  * used to make soil from plants.
+  * Doesn't have components.
+  */
+/obj/machinery/compost_bin
+	name = "compost bin"
+	desc = "A wooden bin for composting"
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "compost_bin-empty"
+	power_state = NO_POWER_USE
+	density = TRUE
+	anchored = TRUE
+	// amount of biomass in the compost bin
+	var/biomass = 0
+	// amount of compost in the compost bin
+	var/compost = 0
+	// The maximum amount of biomass the compost bin can store.
+	var/biomass_capacity = BASE_BIOMASS_CAPACITY
+	// The maximum amount of compost the compost bin can store.
+	var/compost_capacity = BASE_COMPOST_CAPACITY
+	// Is the bin currently composting
+	var/composting = FALSE
+
+/obj/machinery/compost_bin/Initialize(mapload)
+	// try to compost
+	compost()
+	return
```

Comment:
1. You have to call parent on this
2. Won't the bin start out empty so this won't do anything?

Reply:
I wasn't entirely certain whether it would in all cases. It's more of a just in case anyhow. And what exactly do you mean by have to call parent?

Reply:
Okay so, in OOP you can work off of subtypes. I posted an example of it in the discord in #code_testing , but the gist is:
If you define proc A on `obj/A`, then `obj/A/B`, where object B is a subtype of A, can also access proc A.
In this specific case, `/obj` or `/obj/machinery` handles some other stuff that machinery needs in their `Initialize`, which won't be properly set up if you don't "Call parent". In DM you can easily call parent with `..()`, which I think you've seen before!
```
return ..()
```
^^This is what you'd have to add to it!

Reply:
oh, I see.